docs/extending_syzkaller.md

```diff
@@ -0,0 +1,39 @@
+# Extending syzkaller
+
+Here are the common files and some general edits that need to be made in order to extend syzkaller for a new OS kernel. However, there are some specific changes that may be required for a given kernel (For example gathering coverage from a given kernel, or some errors that might pop up and give a hint about what to tweak).
+
+## First general file
+
+`executor/executor.cc`
+
+This file is mainly responsible for executing the syscalls programs, and managing the
+threads in which the programs run. It also contains the `init_os` function which is responsible for mapping a virtual address space for the calling process, and “execute_syscall” which is responsible for executing system calls for a particular OS kernel. `init_os`and `execute_syscall` are implemented in files in which their names follow a pattern `executor_os_name.h` where os_name can be linux, fuschia, windows, karos, etc.
+
+`executor_os_name.h` also contains functions related to that operating system such as functions that allow it to gather coverage information, detect bitness, etc. (Example: [executor_linux.h](/executor/executor_linux.h) ).
+
+The intended function will be called according to the target as defined by the macros in the “executor.cc” file.
```

Comment:
Please use consistent style to refer to file name. Either back-quotes or double-quotes.
I think back-quotes is better.